core-primitives/extrinsics-factory/src/lib.rs

```diff
@@ -0,0 +1,162 @@
+/*
+	Copyright 2021 Integritee AG and Supercomputing Systems AG
+
+	Licensed under the Apache License, Version 2.0 (the "License");
+	you may not use this file except in compliance with the License.
+	You may obtain a copy of the License at
+
+		http://www.apache.org/licenses/LICENSE-2.0
+
+	Unless required by applicable law or agreed to in writing, software
+	distributed under the License is distributed on an "AS IS" BASIS,
+	WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+	See the License for the specific language governing permissions and
+	limitations under the License.
+
+*/
+
+#![cfg_attr(not(feature = "std"), no_std)]
+
+#[cfg(all(feature = "std", feature = "sgx"))]
+compile_error!("feature \"std\" and feature \"sgx\" cannot be enabled at the same time");
+
+#[cfg(all(not(feature = "std"), feature = "sgx"))]
+extern crate sgx_tstd as std;
+
+// re-export module to properly feature gate sgx and regular std environment
+#[cfg(all(not(feature = "std"), feature = "sgx"))]
+pub mod sgx_reexport_prelude {
+	pub use thiserror_sgx as thiserror;
+}
+
+use codec::Encode;
+use error::Result;
+use itp_nonce_cache::{MutateNonce, Nonce};
+use itp_settings::node::{RUNTIME_SPEC_VERSION, RUNTIME_TRANSACTION_VERSION};
+use itp_types::OpaqueCall;
+use sp_core::{Pair, H256};
+use sp_runtime::{MultiSignature, OpaqueExtrinsic};
+use std::{sync::Arc, vec::Vec};
+use substrate_api_client::compose_extrinsic_offline;
+
+pub mod error;
+
+/// Create extrinsics from opaque calls
+///
+/// Also increases the nonce counter for each extrinsic that is created.
+pub trait CreateExtrinsics {
+	fn create_extrinsics(&self, calls: &[OpaqueCall]) -> Result<Vec<OpaqueExtrinsic>>;
+}
```

Comment:
This is the new trait to create extrinsics, transforming `OpaqueCall` into `OpaqueExtrinsic`